pandas/core/arrays/boolean.py

```diff
@@ -182,6 +182,7 @@ def coerce_to_array(
             values = values.copy()
     elif isinstance(values, np.ndarray) and is_numeric_dtype(values.dtype):
         mask_values = isna(values)
+        assert mask_values is not None
```

Comment:
Because of `mask_values = None` at the top of the function, mypy still weirdly beliefs that the type is `None | npt.NDArray[np.bool_]` (pyright infers it correctly - cannot be None).

Reply:
Unfortunately, that still procures the mypy errors.  I think the issue is that mypy does not narrow the type within the different if blocks. After the if-blocks the type is `None | npt.NDArray[np.bool_]` (some if blocks leave `mask_values` as `None`). I will remove the assert and replace it with a `cast`. If a future mypy version can narrow the type, it will tell use to remove the cast.